server/oauth2.go

```diff
@@ -125,6 +135,92 @@ func parseScopes(scopes []string) connector.Scopes {
 	return s
 }
 
+func signatureAlgorithm(jwk *jose.JSONWebKey) (alg jose.SignatureAlgorithm, err error) {
```

Comment:
Would be helpful to write up a brief comment describing the following two functions.